### DISCUSSION TITLE
How to debug "Model was trained with exogenous variables but you have not passed any for predictions. Please pass exogenous variables to make predictions"?

### DISCUSSION THREAD

**Discussion by jbrepogmailcom** [pycaret/pycaret#3544] [Q&A]
Hello. I am trying to predict time series with command

`predictions = predict_model(f_best,fh=(fh+5))`

after I used tune_model() and finalize_model()

However, I get this error:

**Model was trained with exogenous variables but you have not passed any for predictions. Please pass exogenous variables to make predictions**

I only used these commands:

```
fh = 60
s = setup(data, fh = fh, session_id = 123, target = "rate") 
best = compare_models()
t_best = tune_model(best)
f_best = finalize_model(t_best)
predictions = predict_model(f_best,fh=(fh+5)) 
```

How to debug this error, or how do I pass exogenous variables? Thanks very much

**Comment by ngupta23** [comment]
Please check out the tutorials on time series modeling with exogenous variables Section 6: [Link ](https://nbviewer.org/github/pycaret/pycaret/blob/master/tutorials/time_series/forecasting/univariate_with_exogeneous_part2.ipynb).
